packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -286,7 +291,7 @@ class BuildIOSFrameworkCommand extends BuildFrameworkCommand {
 
     globals.printStatus('Frameworks written to ${outputDirectory.path}.');
 
-    if (!project.isModule && hasPlugins(project)) {
+    if (!project.isModule && boolArg('plugins') && hasPlugins(project)) {
```

Comment:
Why is this being gated on the `plugins` flag? This step should be basically free (so wouldn't impact build times, which as I understand it is the driver for this PR), and it's not at all clear to me that it wouldn't be needed in the use cases described here. Isn't the intended outcome that the app still has and uses plugins, they are just built separately? Building all the plugins via some other mechanism doesn't remove the need for a registrant.